src/ophys_etl/modules/segmentation/validation/video_creation.py

```diff
@@ -0,0 +1,344 @@
+            r = int(spacing / 2 + ir * spacing)
+            c = int(spacing / 2 + ic * spacing)
+            rois.append(create_roi_ellipse(center=(r, c),
+                                           r_radius=r_radius,
+                                           c_radius=c_radius,
+                                           rotation=rotation,
+                                           shape=shape[1:],
+                                           id=roi_id))
+            roi_summary[roi_id] = {
+                    "center": (r, c)}
+            roi_id += 1
+
+    # determine a weight factor for each ROI
+    weights = polynomial_weight_mask(rois[0], 2)
+    wfactors = np.linspace(correlation_high,
+                           correlation_low,
+                           len(rois))
+
+    # simulate traces for each pixel in each ROI
+    traces = np.zeros(shape=shape, dtype="uint16")
+    rng = np.random.default_rng(seed=123)
+    t = np.arange(shape[0])
+    for wfactor, roi in zip(wfactors, rois):
+        rng.shuffle(t)
+        magnitudes = rng.random(size=n_events)
+        # for this ROI, simulate a common-mode trace
+        trace = validation.sum_events(n_samples=shape[0],
+                                      timestamps=t[0:n_events],
+                                      magnitudes=magnitudes,
+                                      decay_time=decay_time,
+                                      rate=rate)
+
+        # for every pixel in this ROI, create a trace with a specified
+        # correlation to the common-mode
+        roi_traces = correlated_traces_from_weights(
+                common_trace=trace,
+                weights=(weights * wfactor))
+
+        # for any non-zero trace in this ROI, estimate the noise
+        # and get the minimum value
+        not_empty_indices = np.argwhere(weights != 0)
+        noises = []
+        tmins = []
+        for indices in not_empty_indices:
+            trace = roi_traces[:, indices[0], indices[1]]
+            noises.append(
+                    trace_noise_estimate(trace, filt_length=33))
+            tmins.append(trace.min())
+        noise = np.mean(noises)
+        tmin = np.mean(tmins)
+
+        # convert to uint16 with max value 2**12 and offset 500
+        tmin = roi_traces.min()
+        tptp = roi_traces.ptp()
+        toff = 500
+        roi_traces = ((roi_traces - tmin) * np.power(2, 12) / tptp) + toff
+        roi_traces = roi_traces.astype("uint16")
+        roi_summary[roi["id"]].update(
+                {
+                    "noise": noise,
+                    "tmin": tmin,
+                    "tptp": tptp,
+                    "toff": toff})
+
+        # place the traces for this ROI in the global movie
+        for indices in not_empty_indices:
+            traces[:, roi["y"] + indices[0], roi["x"] + indices[1]] = \
+                    roi_traces[:, indices[0], indices[1]]
+
+    # for all the ROIs, determine the average offsets, noise etc
+    r = []
```

Comment:
It looks like you don't need these `r` and `c` arrays.

Reply:
right, got rid of them.